src/spell.py

```diff
@@ -1,78 +1,97 @@
-
-import sys
-from typing import Dict
+import argparse
 
 law_enforcement_spelling = {
-    'a': 'Adam',
-    'b': 'Boston',
-    'c': 'Charles',
-    'd': 'David',
-    'e': 'Edward',
-    'f': 'Frank',
-    'g': 'George',
-    'h': 'Henry',
-    'i': 'Ida',
-    'j': 'James',
-    'k': 'King',
-    'l': 'Lincoln',
-    'm': 'Mary',
-    'n': 'Nora',
-    'o': 'Oscar',
-    'p': 'Paul',
-    'q': 'Queen',
-    'r': 'Robert',
-    's': 'Sam',
-    't': 'Tom',
-    'u': 'Union',
-    'v': 'Victor',
-    'w': 'William',
-    'x': 'X-Ray',
-    'y': 'Young',
-    'z': 'Zebra',
+    "a": "Adam",
+    "b": "Boston",
+    "c": "Charles",
+    "d": "David",
+    "e": "Edward",
+    "f": "Frank",
+    "g": "George",
+    "h": "Henry",
+    "i": "Ida",
+    "j": "James",
+    "k": "King",
+    "l": "Lincoln",
+    "m": "Mary",
+    "n": "Nora",
+    "o": "Oscar",
+    "p": "Paul",
+    "q": "Queen",
+    "r": "Robert",
+    "s": "Sam",
+    "t": "Tom",
+    "u": "Union",
+    "v": "Victor",
+    "w": "William",
+    "x": "X-Ray",
+    "y": "Young",
+    "z": "Zebra",
 }
 
 nato_spelling = {
-    'a': 'Álfa',
-    'b': 'Brávo',
-    'c': 'Chárlie',
-    'd': 'Délta',
-    'e': 'Écho',
-    'f': 'Fóxtrot',
-    'g': 'Gólf',
-    'h': 'Hotél',
-    'i': 'Índia',
-    'j': 'Júliett',
-    'k': 'Kílo',
-    'l': 'Líma',
-    'm': 'Mike',
-    'n': 'Novémber',
-    'o': 'Óscar',
-    'p': 'Papá',
-    'q': 'Quebéc',
-    'r': 'Rómeo',
-    's': 'Siérra',
-    't': 'Tángo',
-    'u': 'Úniform',
-    'v': 'Víctor',
-    'w': 'Whísky',
-    'x': 'X-ray',
-    'y': 'Yánkee',
-    'z': 'Zúlu'
+    "a": "Álfa",
+    "b": "Brávo",
+    "c": "Chárlie",
+    "d": "Délta",
+    "e": "Écho",
+    "f": "Fóxtrot",
+    "g": "Gólf",
+    "h": "Hotél",
+    "i": "Índia",
+    "j": "Júliett",
+    "k": "Kílo",
+    "l": "Líma",
+    "m": "Mike",
+    "n": "Novémber",
+    "o": "Óscar",
+    "p": "Papá",
+    "q": "Quebéc",
+    "r": "Rómeo",
+    "s": "Siérra",
+    "t": "Tángo",
+    "u": "Úniform",
+    "v": "Víctor",
+    "w": "Whísky",
+    "x": "X-ray",
+    "y": "Yánkee",
+    "z": "Zúlu",
 }
 
+parser = argparse.ArgumentParser(
+    description="A simple command line tool "
+    "to display a spelling for any phrase "
+    "using a phonetic alphabet."
+)
+parser.add_argument(
+    "--alphabet",
+    default="nato",
+    choices=["nato", "law_enforcement"],
+    dest="alphabet",
+)
+parser.add_argument(dest="phrase")
+parser.add_argument("--print-letter", default=1, dest="print_letter")
+
 
-def print_spelling(word: str, spelling_map: Dict[str, str]) -> None:
-    for letter in word:
-        print(letter, ': ', sep='', end='')
+def print_spelling(args: argparse.ArgumentParser) -> None:
+    if args.alphabet == "nato":
+        spelling_map = nato_spelling
+    elif args.alphabet == "law_enforcement":
+        spelling_map = law_enforcement_spelling
+    else:
+        raise ValueError(f"Improper alphabet arg, must be {args.alphabet.choices}")
```

Comment:
This probably isn't necessary (or even reachable), I'm sure there's some way to return error messages if the value passed in isn't part of the `choices=["nato", "law_enforcement"]` param I added to the first argument. I'll clean this up later once I've had a closer look at the argparse docs.

Reply:
Let's keep it here to catch the case when a new alphabet is added.